src/components/MerchantsPage/gam/CampaignProgressBar.tsx

```diff
@@ -0,0 +1,135 @@
+
+  const timeUntilEnd = endDate.getTime() - Date.now();
+  const daysUntilEnd = timeUntilEnd / (1000 * 3600 * 24);
+
+  return (
+    <ProgressBarContainer>
+      {hoursSinceLastContribution < 1 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(minutesSinceLastConstribution)}m
+          ago
+        </TimeStamp>
+      )}
+      {hoursSinceLastContribution >= 1 && hoursSinceLastContribution < 24 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(hoursSinceLastContribution)}h ago
+        </TimeStamp>
+      )}
+      {hoursSinceLastContribution >= 24 && (
+        <TimeStamp>
+          Last contribution made {Math.floor(daysSinceLastContribution)}d ago
+        </TimeStamp>
+      )}
+      <TargetAmountBar className="progress-bar">
+        <CurrentProgressBar
+          style={{
+            width: `${progressWidth(numContributions, targetAmount)}%`,
+            backgroundColor: progressBarColor,
+            //defaults to default color if no color is passed in
+          }}
+        >
+          {' '}
+        </CurrentProgressBar>
+      </TargetAmountBar>
+      <ContributionInfoContainer>
+        <div>
+          <ProgressTextContainer color={progressBarColor}>
+            {progressPercent}% to target
+          </ProgressTextContainer>{' '}
+          ({numContributions} out of {targetAmount} meals)
+        </div>
+        {daysUntilEnd > 1 && (
+          <div>
+            Ends in{' '}
+            <ProgressTextContainer color={progressBarColor}>
+              {Math.ceil(daysUntilEnd)}
+            </ProgressTextContainer>{' '}
+            days
+          </div>
+        )}
+        {daysUntilEnd > 0 && daysUntilEnd <= 1 && (
+          <div>
+            Ends in{' '}
+            <ProgressTextContainer color={progressBarColor}>
+              {Math.ceil(daysUntilEnd)}
+            </ProgressTextContainer>{' '}
+            day
+          </div>
+        )}
+        {daysUntilEnd <= 0 && <div>Ended</div>}
+      </ContributionInfoContainer>
+    </ProgressBarContainer>
+  );
+};
+
+export default CampaignProgressBar;
+
+const ProgressBarContainer = styled.div`
+  width: 100%;
+  padding: 15px 0;
+`;
+
+const ProgressTextContainer = styled.span`
+  font-family: Open Sans;
+  font-style: normal;
+  font-weight: 600;
+  font-size: 14px;
+  line-height: 22px;
+  color: ${(props) => props.color};
+`;
+
+const TargetAmountBar = styled.div`
+  background-color: #dedede;
+  height: 12px;
+  margin-bottom: 15px;
+`;
+
+const CurrentProgressBar = styled.div`
+  background-color: #dd678a;
+  z-index: 5;
+  height: 12px;
+`;
+
+const ContributionInfoContainer = styled.div`
+  display: flex;
+  justify-content: space-between;
+`;
+
+const TimeStamp = styled.div`
+  font-family: Open Sans;
+  font-style: normal;
+  font-weight: normal;
+  font-size: 13px;
+  line-height: 18px;
+  letter-spacing: 0.02em;
+  color: #9e9e9e;
```

Comment:
Are these colors not stored in a constant somewhere?

Reply:
I put some colors to be shared in `src/index.scss`. I think that's a good place to be sharing these if you want to put this and/or others there.

Reply:
Done! Added these colors to src/index.scss. I think we would need additional eng work to export the color constants to be available in components, but this is a good start